### ISSUE TITLE
Default identity files are not used when private key file is not provided

### ISSUE THREAD

**Issue by binooetomo** [ParallelSSH/parallel-ssh#222]
Dear All

I have target host that currently I can ssh to it without password.
But I fail to do it using parallel-ssh.

```
(coba2) debian@debian-mon:~/coba2$ ssh root@10.90.200.55 -p 3322
Last login: Fri Sep 18 10:23:25 2020 from 10.90.200.57
[root@candibinangun ~]# exit
logout
Connection to 10.90.200.55 closed.

(coba2) debian@debian-mon:~/coba2$ python
Python 3.8.5 (default, Aug 28 2020, 06:44:01) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pssh.clients import SSHClient
>>> sshObj=SSHClient('10.90.200.55', user='root', port=3322, num_retries=3, retry_delay=5, timeout=5)
Traceback (most recent call last):
  File "/home/debian/.local/share/virtualenvs/coba2-gQE_YhKQ/lib/python3.8/site-packages/pssh/clients/base/single.py", line 93, in _auth_retry
    self.auth()
  File "/home/debian/.local/share/virtualenvs/coba2-gQE_YhKQ/lib/python3.8/site-packages/pssh/clients/native/single.py", line 179, in auth
    self._identity_auth()
  File "/home/debian/.local/share/virtualenvs/coba2-gQE_YhKQ/lib/python3.8/site-packages/pssh/clients/base/single.py", line 183, in _identity_auth
    raise AuthenticationException("No authentication methods succeeded")
pssh.exceptions.AuthenticationException: No authentication methods succeeded

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/debian/.local/share/virtualenvs/coba2-gQE_YhKQ/lib/python3.8/site-packages/pssh/clients/base/single.py", line 93, in _auth_retry
    self.auth()
  File "/home/debian/.local/share/virtualenvs/coba2-gQE_YhKQ/lib/python3.8/site-packages/pssh/clients/native/single.py", line 179, in auth
    self._identity_auth()
  File "/home/debian/.local/share/virtualenvs/coba2-gQE_YhKQ/lib/python3.8/site-packages/pssh/clients/base/single.py", line 183, in _identity_auth
    raise AuthenticationException("No authentication methods succeeded")
pssh.exceptions.AuthenticationException: No authentication methods succeeded

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/debian/.local/share/virtualenvs/coba2-gQE_YhKQ/lib/python3.8/site-packages/pssh/clients/base/single.py", line 93, in _auth_retry
    self.auth()
  File "/home/debian/.local/share/virtualenvs/coba2-gQE_YhKQ/lib/python3.8/site-packages/pssh/clients/native/single.py", line 179, in auth
    self._identity_auth()
  File "/home/debian/.local/share/virtualenvs/coba2-gQE_YhKQ/lib/python3.8/site-packages/pssh/clients/base/single.py", line 183, in _identity_auth
    raise AuthenticationException("No authentication methods succeeded")
pssh.exceptions.AuthenticationException: No authentication methods succeeded

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/debian/.local/share/virtualenvs/coba2-gQE_YhKQ/lib/python3.8/site-packages/pssh/clients/native/single.py", line 103, in __init__
    super(SSHClient, self).__init__(
  File "/home/debian/.local/share/virtualenvs/coba2-gQE_YhKQ/lib/python3.8/site-packages/pssh/clients/base/single.py", line 80, in __init__
    self._init()
  File "/home/debian/.local/share/virtualenvs/coba2-gQE_YhKQ/lib/python3.8/site-packages/pssh/clients/base/single.py", line 85, in _init
    self._auth_retry()
  File "/home/debian/.local/share/virtualenvs/coba2-gQE_YhKQ/lib/python3.8/site-packages/pssh/clients/base/single.py", line 97, in _auth_retry
    return self._auth_retry(retries=retries+1)
  File "/home/debian/.local/share/virtualenvs/coba2-gQE_YhKQ/lib/python3.8/site-packages/pssh/clients/base/single.py", line 97, in _auth_retry
    return self._auth_retry(retries=retries+1)
  File "/home/debian/.local/share/virtualenvs/coba2-gQE_YhKQ/lib/python3.8/site-packages/pssh/clients/base/single.py", line 99, in _auth_retry
    raise AuthenticationException(msg, self.host, self.port, ex)
pssh.exceptions.AuthenticationException: ('Authentication error while connecting to %s:%s - %s', '10.90.200.55', 3322, AuthenticationException('No authentication methods succeeded'))

```

Did I miss somethings?
Kindly please give me any clue to fix this problem.

Sincerely
-bino-

**Comment by pkittenis** [comment]
See [documentation on agents and private keys](https://parallel-ssh.readthedocs.io/en/latest/advanced.html#agents-and-private-keys).

If a valid private key does not exist on an agent, the client will need to provide a valid key programmatically. A set of default keys are attempted if none are provided, [list of which can be found here](https://parallel-ssh.readthedocs.io/en/latest/base_single.html#pssh.clients.base.single.BaseSSHClient.IDENTITIES).

OpenSSH command line utilities can have configuration outside the scope of the library.

If there is a valid key on an agent, (`ssh-add -l`), or a private key path, that works with openssh but not the library please show debug output from client and can re-open.

**Comment by iivanou** [comment]
This issue is not related to valid/invalid keys. The problem is that when you are not specifying an auth key, the `_auth_pkey()` method [is being called](https://github.com/ParallelSSH/parallel-ssh/blob/master/pssh/clients/base/single.py#L302) within the `_identity_auth` method of the BaseSSHClient class with 2 arguments instead of one. 

I've added notification about an exception to logger and got the following output:
```
Trying to authenticate with identity file /root/.ssh/id_rsa
Traceback (most recent call last):
  File ".../python3.6/site-packages/pssh/clients/base/single.py", line 303, in _identity_auth
    self._pkey_auth(identity_file, password=self.password)
TypeError: _pkey_auth() got multiple values for argument 'password'
```

**Comment by pkittenis** [comment]
Thanks for the report - resolved in ``2.5.0``.

As always, code to reproduce and debug output is very helpful to diagnose issues. See issue template for instructions.
